component/component/vshn_redis.jsonnet

```diff
@@ -509,6 +509,12 @@ local composition =
                 controlNamespace: params.services.controlNamespace,
                 restoreSA: 'redisrestoreserviceaccount',
                 quotasEnabled: std.toString(params.services.vshn.quotasEnabled),
+                emailAlertingEnabled: std.toString(params.services.vshn.emailAlerting.enabled),
```

Comment:
I know I start to sound like a broken record in all my reviews, but maybe a function to generate these configs will avoid copy/pasting in the future.

Reply:
Can you clarify what exactly you mean? How does a function help here? It's only passing a parameter to the configMap for the xfn container

Reply:
We have the same `emailAlerting*` lines repeated over and over for each service. Each time copy pasted, this is bad practice for various reasons:

* Copy paste errors happen
* We would need to change all files again if there are changes

If we instead just have a jsonnet function where we pass `params.services.vshn.emailAlerting` and it returns all those lines, we would make it much less error prone to add a new service with email alerting. The we just need to call the function here and we're done. If there are any changes in the future we'd only have to adjust the function and all services will get the changes.

Reply:
Ah I understand now. Refactored the code accordingly. Please check again.